### ISSUE TITLE
// character sequence breaks convar/concommand strings

### ISSUE THREAD

**Issue by TiberiumFusion** [Facepunch/garrysmod-issues#5014]
### Details
Using `//` in a string for a convar or concommand breaks the string. The `//` and all following characters are ignored and thus the convar/concommand receives incorrect data. This can be worked around by deliberately wrapping arguments in `"` quotes.

### Steps to reproduce
```
concommand.Add("testcommand",
	function(p, cmdString, args, argStr)
		print("argStr: ", argStr)
		print("argStr len: ", #argStr)
		print("args --v")
		PrintTable(args)
	end)
```
- `testcommand hello` becomes `hello`
- `testcommand hello/world` becomes `hello/world`
- `testcommand hello//world` becomes `hello`
- `testcommand hello///world` becomes `hello`
- `testcommand https://hello.world` becomes `https:`
- `testcommand 'hello//world'` becomes `'hello`
- `testcommand "hello//world"` becomes `"hello//world"`

### Environment
2021.07.12

This issue has been around for a long time. Maybe it's just Source being Source. Maybe it has existed since HL2. But it is obnoxious nonetheless and seems like a strange quirk that could stand to be fixed.

**Comment by sanny-io** [comment]
Does it happen if you wrap it in quotes?

`testcommand "https://hello.world"`

**Comment by TiberiumFusion** [comment]
@sanny-io The issue does not occur when wrapping the argument(s) in quotes. Yes, this is fine for invoking concommands from Lua, yes, but that misses the really big point here.

The average player is not going to use quotes when typing something into their console. This is especially true for cvars or commands that expect a single argument, e.g. `default_url https://some.url`. Even if you take every step possible to remind the user to step out of their comfort zone and add those super _weird_ `"`  things when they type, they can easily (and likely) ignore/forget/defy your instructions.

I do not know why `//` terminates the concommand string, but it's very unusual and very few players are aware of it. Unless there is some forsaken chunk of ancient Source code that absolutely depends on the current behavior, I don't see why this issue shouldn't be fixed.

**Comment by sanny-io** [comment]
I agree with you. This actually appears in other Source games. I think I've got this figured out:

`//` is used for comments, such as in .cfg files.
